oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/OAuth2UserAuthority.java

```diff
@@ -85,13 +86,13 @@ public boolean equals(Object obj) {
 		if (!this.getAuthority().equals(that.getAuthority())) {
 			return false;
 		}
-		return this.getAttributes().equals(that.getAttributes());
+		return convertURLs(this.getAttributes()).equals(convertURLs(that.getAttributes()));
```

Comment:
We should try to optimize the `equals()` and `hashCode()` since it creates a new `Map` on each call. One option would be to introduce a member variable `private final Map<String, Object> convertedAttributes` that is created at construction time and used in `equals()` and `hashCode()` or an internal `Comparator`  instance. I think I prefer the latter.

Reply:
Hi,

sure, if you think it's worth the complexity. Sorry for the delay, I was in my COVIDs this week :)

I first tried the simple approach of just adding a `private final Map<String, Object> convertedAttributes` but got failing tests from
```
OAuth2AuthenticationTokenMixinTests.serializeWhenMixinRegisteredThenSerializes()
OAuth2AuthenticationTokenMixinTests.serializeWhenRequiredAttributesOnlyThenSerializes()
```
I guess it's forbidden for some reason to add new members to this class?

I'm afraid I didn't understand the suggestion to use a `Comparator`. It wouldn't help with hashCode and there's no need for natural ordering, so wouldn't a `Comparator` just bring additional complexity?

Here's the updated PR. I had to duplicate equals/hashCode functionality of a map. I feel this is somewhat complicated, please let me know if I missed some obvious simple implementation.